src/frontend/screens/ManualGpsScreen/DmsForm/index.js

```diff
@@ -0,0 +1,243 @@
+  },
+  longitude: {
+    id: "screens.ManualGpsScreen.DmsForm.longitude",
+    defaultMessage: "Longitude",
+  },
+  north: {
+    id: "screens.ManualGpsScreen.DmsForm.north",
+    defaultMessage: "North",
+  },
+  south: {
+    id: "screens.ManualGpsScreen.DmsForm.south",
+    defaultMessage: "South",
+  },
+  east: {
+    id: "screens.ManualGpsScreen.DmsForm.east",
+    defaultMessage: "East",
+  },
+  west: {
+    id: "screens.ManualGpsScreen.DmsForm.west",
+    defaultMessage: "West",
+  },
+});
+
+const DIRECTION_OPTIONS_NORTH_SOUTH = [
+  {
+    value: "N",
+    label: m.north.defaultMessage,
+  },
+  {
+    value: "S",
+    label: m.south.defaultMessage,
+  },
+];
+
+const DIRECTION_OPTIONS_EAST_WEST = [
+  {
+    value: "E",
+    label: m.east.defaultMessage,
+  },
+  {
+    value: "W",
+    label: m.west.defaultMessage,
+  },
+];
+
+export type DmsData = {|
+  degrees: string,
+  minutes: string,
+  seconds: string,
+|};
+
+export type DmsUnit = $Keys<DmsData>;
+
+const DmsForm = ({ location, onValueUpdate }: FormProps) => {
+  const [latitude, setLatitude] = React.useState<DmsData>(() => {
+    if (!location.savedPosition)
+      return {
+        degrees: "",
+        minutes: "",
+        seconds: "",
+      };
+
+    const { latitude } = location.savedPosition.coords;
+    const {
+      raw: { degrees, minutes, seconds },
+    } = toDegreesMinutesAndSeconds(latitude);
+
+    return {
+      degrees: degrees.toString(),
+      minutes: minutes.toString(),
+      seconds: seconds.toString(),
+    };
+  });
+  const [longitude, setLongitude] = React.useState<DmsData>(() => {
+    if (!location.savedPosition)
+      return {
+        degrees: "",
+        minutes: "",
+        seconds: "",
+      };
+
+    const { longitude } = location.savedPosition.coords;
+    const {
+      raw: { degrees, minutes, seconds },
+    } = toDegreesMinutesAndSeconds(longitude);
+
+    return {
+      degrees: degrees.toString(),
+      minutes: minutes.toString(),
+      seconds: seconds.toString(),
+    };
+  });
+
+  const [latCardinality, setLatCardinality] = React.useState(
+    getInitialCardinality("lat", location)
+  );
+  const [lonCardinality, setLonCardinality] = React.useState(
+    getInitialCardinality("lon", location)
+  );
+
+  const updateCoordinate = (field: "lat" | "lon") => (
+    unit: DmsUnit,
+    value: string
+  ) => {
+    const update = field === "lat" ? setLatitude : setLongitude;
+
+    // Ideally would use computed property syntax to avoid this but Flow is being annoying
+    update(previous => {
+      switch (unit) {
+        case "degrees":
+          return {
+            ...previous,
+            degrees: value,
+          };
+        case "minutes":
+          return {
+            ...previous,
+            minutes: value,
+          };
+        case "seconds":
+          return {
+            ...previous,
+            seconds: value,
+          };
+        default:
+          return previous;
+      }
+    });
+  };
```

Comment:
ideally this could've just been something like `update(previous => ({...previous, [unit]: value})` but Flow was complaining. I've done this in TS without much issue so a little unfortunate, but this is pretty explicit so could be worse 🤷‍♂️

Reply:
How much time do you think it would take to refactor these flow pages in to ts pages?

Reply:
Wouldn't take long but just didn't want to worry about setting up TS, which you're handling in https://github.com/digidem/mapeo-mobile/pull/634